spoon/cookie/cookie.php

```diff
@@ -114,10 +114,29 @@ public static function get($key)
 		$value = (get_magic_quotes_gpc()) ? stripslashes($_COOKIE[$key]) : $_COOKIE[$key];
 
 		// unserialize
-		$actualValue = @unserialize($value);
+		$actualValue = json_decode($value);
+
+		// if the decode failed
+		if ($actualValue === null) {
+			// maybe it's serialized, check if there are any serialized objects in the string.
+			preg_match('/O:\d+:"/', $value, $matches);
+
+			// no objects were found
+			if (empty($matches)) {
+				$unserializedValue = unserialize($value);
+
+				// unserialize was successful
+				if ($unserializedValue !== false) {
```

Comment:
you need to add an addition check here that checks that the serialized value isn't false otherwise it will get lost

Reply:
I've fixed this in 5cc8233